modules/ppcp-wc-gateway/src/Helper/PayUponInvoiceHelper.php

```diff
@@ -77,6 +77,39 @@ public function is_checkout_ready_for_pui(): bool {
 		return true;
 	}
 
+	/**
+	 * Checks whether pay for order is ready for PUI.
+	 *
+	 * @return bool
+	 */
+	public function is_pay_for_order_ready_for_pui(): bool {
+		/**
+		 * Needed for WordPress `query_vars`.
+		 *
+		 * @psalm-suppress InvalidGlobal
+		 */
+		global $wp;
+
+		$order_id = isset( $wp->query_vars['order-pay'] ) ? (int) $wp->query_vars['order-pay'] : 0;
+		if ( $order_id === 0 ) {
+			return false;
+		}
+
+		$order = wc_get_order( $order_id );
+		if ( ! is_a( $order, WC_Order::class ) ) {
+			return false;
+		}
+
+		$address = $order->get_address();
```

Comment:
Does it return only required fields?

Reply:
yup, `get_address` returned fields are required for creating the payment source: https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-wc-gateway/src/Gateway/PayUponInvoice/PaymentSourceFactory.php#L47-L56

Reply:
I mean it returns something like this (for a normal checkout order)

```
{
  ["first_name"]=> string(4) "John"
  ["last_name"]=> string(3) "Doe"
  ["company"]=> string(0) ""
  ["address_1"]=> string(9) "1 Main St"
  ["address_2"]=> string(2) "12"
  ["city"]=> string(8) "San Jose"
  ["state"]=> string(0) ""
  ["postcode"]=> string(5) "11111"
  ["country"]=> string(2) "DE"
  ["email"]=>w string(37) "...@personal.example.com"
  ["phone"]=> string(10) "4085409578"
}
```

so there is at least empty `state` (not shown for DE in checkout I think) and `company`. Maybe other plugins can also add fields.

Seems like the `''` check will always return `false`?

Reply:
Fixed, thanks :)